pubsublite/producer.go

```diff
@@ -177,17 +184,16 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 		if err != nil {
 			return fmt.Errorf("pubsublite: failed to get publisher: %w", err)
 		}
+		if !p.cfg.Sync {
+			ctx = queuecontext.DetachedContext(ctx)
+		}
```

Comment:
I fixed this along the way, otherwise spans may not be ended and / or metrics recorded. We do the same for Kafka:

https://github.com/elastic/apm-queue/blob/f95dfa46df6ff7867f26b33ddd79ee36be153ea9/kafka/producer.go#L218-L221